lib/config/configuration.go

```diff
@@ -1208,17 +1209,41 @@ func applyWindowsDesktopConfig(fc *FileConfig, cfg *service.Config) error {
 	}
 	ldapPassword, err := os.ReadFile(fc.WindowsDesktop.LDAP.PasswordFile)
 	if err != nil {
-		return trace.WrapWithMessage(err, "loading LDAP password from file %v",
+		return trace.WrapWithMessage(err, "error loading LDAP password from file %v",
 			fc.WindowsDesktop.LDAP.PasswordFile)
 	}
+
+	// If a CA file is provided but InsecureSkipVerify is also set to true, throw an
+	// error to make sure the user isn't making a critical security mistake (i.e. thinking
+	// that their LDAPS connection is being verified to be with the CA provided, but it isn't
+	// due to InsecureSkipVerify == true ).
+	if fc.WindowsDesktop.LDAP.DEREncodedCAFile != "" && fc.WindowsDesktop.LDAP.InsecureSkipVerify {
+		return trace.BadParameter("a CA file was provided but insecure_skip_verify was also set to true; confirm that you really want CA verification to be skipped by deleting or commenting-out the der_ca_file configuration value")
+	}
+
+	var cert *x509.Certificate
+	if !fc.WindowsDesktop.LDAP.InsecureSkipVerify && fc.WindowsDesktop.LDAP.DEREncodedCAFile != "" {
```

Comment:
Simplify? We shouldn't care about InsecureSkipVerify here.

```suggestion
	if fc.WindowsDesktop.LDAP.DEREncodedCAFile != "" {
```